tests/st/harness/core/test_runner.py

```
@@ -232,185 +256,237 @@ def _compile_for_cache(test_case: "PTOTestCase", work_dir: Path, dump_passes: bo
+        _compile_for_cache(tc, work_dir, dump_passes)
+        from pypto.runtime.device_runner import compile_and_assemble  # noqa: PLC0415
+
+        chip_callable, runtime_name = compile_and_assemble(work_dir, resolved, pto_isa_commit=pto_isa_commit)
+        return CompileArtifact(
+            work_dir=work_dir,
+            resolved_platform=resolved,
+            error=None,
+            runtime_name=runtime_name,
+            chip_callable=chip_callable,
+        )
+    except Exception as exc:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)